ipatests/test_integration/test_installation.py

```diff
@@ -567,3 +567,71 @@ def test_KRA_install_after_cert_renew(self):
         self.master.run_command(['kinit', 'admin'], stdin_text=passwd)
         cmd = self.master.run_command(['ipa-kra-install', '-p', dm_pass, '-U'])
         self.master.run_command(['systemctl', 'start', 'chronyd'])
+
+
+class TestMaskMasterInstall(IntegrationTest):
+    """ Test master installation with wrong mask
+
+    This test checks that master installation fails (expectedly) if
+    mask > 022.
+
+    related ticket: https://pagure.io/freeipa/issue/7193
+    """
+
+    num_replicas = 0
+
+    @classmethod
+    def install(cls, mh):
+        super(TestMaskMasterInstall, cls).install(mh)
+        cls.bashrc_file = cls.master.get_file_contents('/root/.bashrc')
+
+    def test_install_master(self):
+        self.master.run_command('echo "umask 0027" >> /root/.bashrc')
+        result = self.master.run_command(['umask'])
+        assert '0027' in result.stdout_text
+
+        cmd = tasks.install_master(self.master, raiseonerr=False)
+        exp_str = ("Unexpected system mask")
+        assert (exp_str in cmd.stderr_text and cmd.returncode != 0)
+
+    def test_files_ownership_and_permission_teardown(self):
+        """ Method to restore the default bashrc contents"""
+        if self.bashrc_file is not None:
+            self.master.put_file_contents('/root/.bashrc', self.bashrc_file)
+
+
+class TestUmaskReplicaInstall(IntegrationTest):
+    """ Test replica installation with wrong mask
+
+    This test checks that replica installation fails (expectedly) if
+    mask > 022.
+
+    related ticket: https://pagure.io/freeipa/issue/7193
+    """
+
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        super(TestUmaskReplicaInstall, cls).install(mh)
+        cls.bashrc_file = cls.replicas[0].get_file_contents('/root/.bashrc')
+
+    def test_install_master(self):
+        tasks.install_master(self.master, setup_dns=False)
+
+    def test_install_replica(self):
+        self.replicas[0].run_command('echo "umask 0027" >> /root/.bashrc')
+        result = self.replicas[0].run_command(['umask'])
+        assert '0027' in result.stdout_text
+
+        cmd = tasks.install_replica(
+            self.master, self.replicas[0],
+            setup_dns=False, setup_ca=False, raiseonerr=False)
+        exp_str = ("Unexpected system mask")
+        assert (exp_str in cmd.stderr_text and cmd.returncode != 0)
```

Comment:
The test doesn't need to install a master to call ipa-replica-install (as the command should exit even before the master is contacted). Just use 
self.master.run_command(['ipa-replica-install', '-w',
                                     self.master.config.admin_password,
                                     '-n', self.master.domain.name,
                                     '-r', self.master.domain.realm,
                                     '--server', 'dummy_master.%s'.format(self.master.domain.name),
                                     '-U'])
and set num_replicas=0, this will use only the minimal number of resources.

Reply:
Thanks, I'm merging both test cases into one with your suggestion.